In [1]:
#3
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

#4
%matplotlib inline
#5
gdp_df=pd.read_csv('../data/gdp_percapita.csv')
gdp_df.head()

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2019  2065.036398              NaN
1     Afghanistan  2018  2033.779002              NaN
2     Afghanistan  2017  2058.383832              NaN
3     Afghanistan  2016  2057.062164              NaN
4     Afghanistan  2015  2068.265904              NaN

In [2]:
#6

internet_df=pd.read_csv('../data/internet_use.csv',nrows=4496)
internet_df.head()

Country or Area  Year     Value  Value Footnotes
0     Afghanistan  2014  6.390000              NaN
1     Afghanistan  2013  5.900000              NaN
2     Afghanistan  2012  5.454545              NaN
3     Afghanistan  2011  5.000000              NaN
4     Afghanistan  2010  4.000000              NaN

In [3]:
#7

print(gdp_df.shape)

(6870, 4)


Answer: The shape of the internet use dataset is 6,870 rows and 4 columns

In [4]:
#7 Continued

print(internet_df.shape)

(4496, 4)


Answer: The shape of the internet use dataset is 4,496 rows and 4 columns

In [5]:
#8

print(gdp_df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6870 entries, 0 to 6869
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  6870 non-null   object 
 1   Year             6870 non-null   object 
 2   Value            6868 non-null   float64
 3   Value Footnotes  1 non-null      float64
dtypes: float64(2), object(2)
memory usage: 214.8+ KB
None


In [6]:
#8 Continued

print(internet_df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4496 entries, 0 to 4495
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  4496 non-null   object 
 1   Year             4496 non-null   object 
 2   Value            4495 non-null   float64
 3   Value Footnotes  948 non-null    float64
dtypes: float64(2), object(2)
memory usage: 140.6+ KB
None


In [7]:
#9
print(gdp_df.tail(10))  

     Country or Area                                      Year        Value  \
6860        Zimbabwe                                      1997  3580.048793   
6861        Zimbabwe                                      1996  3520.430146   
6862        Zimbabwe                                      1995  3226.413930   
6863        Zimbabwe                                      1994  3263.934978   
6864        Zimbabwe                                      1993  3033.504852   
6865        Zimbabwe                                      1992  3054.889178   
6866        Zimbabwe                                      1991  3426.598094   
6867        Zimbabwe                                      1990  3324.348171   
6868   footnoteSeqID                                  Footnote          NaN   
6869               2  Excludes South Sudan after July 9, 2011.          NaN   

      Value Footnotes  
6860              NaN  
6861              NaN  
6862              NaN  
6863              NaN  
6864      

In [8]:
#9 Continued

print(internet_df.tail(10)) 

     Country or Area      Year     Value  Value Footnotes
4486        Zimbabwe      2001  0.799846              NaN
4487        Zimbabwe      2000  0.401434              NaN
4488        Zimbabwe      1999  0.161676              NaN
4489        Zimbabwe      1998  0.081648              NaN
4490        Zimbabwe      1997  0.033080              NaN
4491        Zimbabwe      1996  0.016790              NaN
4492        Zimbabwe      1995  0.007684              NaN
4493        Zimbabwe      1994  0.001739              NaN
4494        Zimbabwe      1990  0.000000              NaN
4495   footnoteSeqID  Footnote       NaN              NaN


In [9]:
#10.

del gdp_df['Value Footnotes']
print(gdp_df.head())

  Country or Area  Year        Value
0     Afghanistan  2019  2065.036398
1     Afghanistan  2018  2033.779002
2     Afghanistan  2017  2058.383832
3     Afghanistan  2016  2057.062164
4     Afghanistan  2015  2068.265904


In [10]:
#10 Continued

del internet_df['Value Footnotes']
print(internet_df.head())

  Country or Area  Year     Value
0     Afghanistan  2014  6.390000
1     Afghanistan  2013  5.900000
2     Afghanistan  2012  5.454545
3     Afghanistan  2011  5.000000
4     Afghanistan  2010  4.000000


In [11]:
#11
gdp_df.columns = ["Country", "Year", "GDP_Per_Capita"]
gdp_df.head()

Country  Year  GDP_Per_Capita
0  Afghanistan  2019     2065.036398
1  Afghanistan  2018     2033.779002
2  Afghanistan  2017     2058.383832
3  Afghanistan  2016     2057.062164
4  Afghanistan  2015     2068.265904

In [12]:
#12
internet_df.columns = ["Country", "Year", "Internet_Users_Pct"]
internet_df.head()

Country  Year  Internet_Users_Pct
0  Afghanistan  2014            6.390000
1  Afghanistan  2013            5.900000
2  Afghanistan  2012            5.454545
3  Afghanistan  2011            5.000000
4  Afghanistan  2010            4.000000

In [13]:
#13 & 14
gdp_and_internet_use = pd.merge(gdp_df, internet_df, how='inner', on=["Country", "Year"])
gdp_and_internet_use.head()

Country  Year  GDP_Per_Capita  Internet_Users_Pct
0  Afghanistan  2014     2102.385234            6.390000
1  Afghanistan  2013     2116.465454            5.900000
2  Afghanistan  2012     2075.491747            5.454545
3  Afghanistan  2011     1904.559799            5.000000
4  Afghanistan  2010     1957.029338            4.000000

In [14]:
#15
gdp_and_internet_use = gdp_and_internet_use.iloc[:-1]
print(gdp_and_internet_use)


          Country  Year  GDP_Per_Capita  Internet_Users_Pct
0     Afghanistan  2014     2102.385234            6.390000
1     Afghanistan  2013     2116.465454            5.900000
2     Afghanistan  2012     2075.491747            5.454545
3     Afghanistan  2011     1904.559799            5.000000
4     Afghanistan  2010     1957.029338            4.000000
...           ...   ...             ...                 ...
3453     Zimbabwe  1997     3580.048793            0.033080
3454     Zimbabwe  1996     3520.430146            0.016790
3455     Zimbabwe  1995     3226.413930            0.007684
3456     Zimbabwe  1994     3263.934978            0.001739
3457     Zimbabwe  1990     3324.348171            0.000000

[3458 rows x 4 columns]


In [15]:
#16 
years = gdp_and_internet_use[(gdp_and_internet_use['Year']=='2004') | (gdp_and_internet_use['Year']=='2009') | (gdp_and_internet_use['Year']=='2014')]
print(years)

          Country  Year  GDP_Per_Capita  Internet_Users_Pct
0     Afghanistan  2014     2102.385234            6.390000
5     Afghanistan  2009     1758.904043            3.550000
10    Afghanistan  2004     1200.278256            0.105809
13        Albania  2014    11623.866679           60.100000
18        Albania  2009    10346.864199           41.200000
...           ...   ...             ...                 ...
3419       Zambia  2009     2917.516220            6.310000
3424       Zambia  2004     2261.766537            2.013550
3436     Zimbabwe  2014     2955.240287           19.890000
3441     Zimbabwe  2009     1925.364381           11.360000
3446     Zimbabwe  2004     2480.859342            6.564045

[482 rows x 4 columns]


In [16]:
#17

year2004_df = years[years['Year'] == '2004']
print(year2004_df)

year2009_df = years[years['Year'] == '2009']
print(year2009_df)

year2014_df = years[years['Year'] == '2014']
print(year2014_df)


                  Country  Year  GDP_Per_Capita  Internet_Users_Pct
10            Afghanistan  2004     1200.278256            0.105809
23                Albania  2004     7604.838002            2.420388
44                Algeria  2004    10058.435939            4.634475
66                 Angola  2004     5593.241537            0.464815
86    Antigua and Barbuda  2004    18745.681534           24.266544
...                   ...   ...             ...                 ...
3361              Uruguay  2004    12479.966871           17.063098
3383           Uzbekistan  2004     3221.189333            2.593725
3404              Vanuatu  2004     2792.356252            4.746603
3424               Zambia  2004     2261.766537            2.013550
3446             Zimbabwe  2004     2480.859342            6.564045

[157 rows x 4 columns]
                  Country  Year  GDP_Per_Capita  Internet_Users_Pct
5             Afghanistan  2009     1758.904043            3.550000
18                Albani

In [17]:
#18
country_max_perc_2014 = year2014_df[year2014_df['Internet_Users_Pct'] == year2014_df['Internet_Users_Pct'].max()]
print(country_max_perc_2014)

      Country  Year  GDP_Per_Capita  Internet_Users_Pct
1381  Iceland  2014      49652.1495               98.16


#18 Answer:       
    Iceland had the highest percentage of internet users in 2014. The percentage was 98.16.

In [18]:
#19
country_min_perc_2014 = year2014_df[year2014_df['Internet_Users_Pct'] == year2014_df['Internet_Users_Pct'].min()]
print(country_min_perc_2014)

          Country  Year  GDP_Per_Capita  Internet_Users_Pct
3087  Timor-Leste  2014      3264.56807                1.14


#19 Answer:
Timor-Leste had the lowest percentage of internet users for 2014 at 1.14%

In [19]:
#20
country_min_perc_2004 = year2004_df[year2004_df['Internet_Users_Pct'] == year2004_df['Internet_Users_Pct'].min()]
print(country_min_perc_2004)

country_min_perc_2009 = year2009_df[year2009_df['Internet_Users_Pct'] == year2009_df['Internet_Users_Pct'].min()]
print(country_min_perc_2009)


      Country  Year  GDP_Per_Capita  Internet_Users_Pct
2132  Myanmar  2004      1696.88176            0.024337
          Country  Year  GDP_Per_Capita  Internet_Users_Pct
3092  Timor-Leste  2009     2724.770973            0.185252


#20 Answer: 

Minimum internet usage percentage for 2004 was .02% in the country of Myanmar, but for 2009 it was .18% in Timor-Leste

In [20]:
#21
country_max_gdp_2014 = year2014_df[year2014_df['GDP_Per_Capita'] == year2014_df['GDP_Per_Capita'].max()]
print(country_max_gdp_2014)

         Country  Year  GDP_Per_Capita  Internet_Users_Pct
1798  Luxembourg  2014    108760.91603               94.67


#21 Answer:
Luxembourg had the highest gdp per capita for 2014 at 108760.91603 

In [21]:
country_min_gdp_2014 = year2014_df[year2014_df['GDP_Per_Capita'] == year2014_df['GDP_Per_Capita'].min()]
print(country_min_gdp_2014)

     Country  Year  GDP_Per_Capita  Internet_Users_Pct
540  Burundi  2014      886.235777                1.38


#22 Answer:
Burundi had the lowest gdp per capita for 2014 at 886.235777 